### DISCUSSION TITLE
Project template for adding textcat / multi_label_textcat to a pre-trained model

### DISCUSSION THREAD

**Discussion by balachander1964** [explosion/spaCy#7882] [Help: Coding & Implementations] [answered]
Hi, Could you please let me know how to add the multi_label_textcat pipeline component to a pre-trained NLP model like en_core_web_lg? I request you to let know how to set the configuration parameters. The current version of en_core_web_lg does not contain multi_label_textcat (and even the textcat) pipeline component. Awaiting your reply. Bala

**Comment by svlandeg** [comment, marked as answer]
Hi!

To get a default configuration for the `textcat_multilabel` component, you can run 
```
spacy init config textcat_multi.cfg -p ["textcat_multilabel"]
```
Which will give you something like this:
```
...
[components.textcat_multilabel]
factory = "textcat_multilabel"
threshold = 0.5

[components.textcat_multilabel.model]
@architectures = "spacy.TextCatBOW.v2"
exclusive_classes = false
ngram_size = 1
no_output_layer = false
nO = null
...
```
You can use this block in your [config file](https://spacy.io/usage/training#config) and [source](https://spacy.io/usage/processing-pipelines#sourced-components) the existing components from `en_core_web_lg`, e.g.
```
[components.parser]
source = "en_core_web_lg"
component = "parser"
```

If you're not using the config system, but compiling the `nlp` object in code, you can do something like:
```
from spacy.pipeline.textcat_multilabel import DEFAULT_MULTI_TEXTCAT_MODEL
nlp = spacy.load("en_core_web_lg")
config = {
    "threshold": 0.5,
    "model": DEFAULT_MULTI_TEXTCAT_MODEL,
    }
nlp.add_pipe("textcat_multilabel", config=config)
```
You can tune the parameter settings according to your needs/data. But the default settings are usually reasonable starting values for experimentation.

**Reply by balachander1964** [reply]
Thank you very much. Let me try this.

**Reply by balachander1964** [reply]
Hi svlandeg, It worked. I appreciate your help just in time.

**Reply by svlandeg** [reply]
Happy to hear it!

**Comment by balachander1964** [comment]
Thank you very much. Let me try this.

**Comment by balachander1964** [comment]
Hi, Thank you very much. let me try this.
